### ISSUE TITLE
Addition of new dependency updates all already installed

### ISSUE THREAD

**Issue by tribals** [python-poetry/poetry#2995]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Solus Linux 4.1
- **Poetry version**: 1.0.10
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: 

## Issue

Hello.

Right now. adding new dependency with `poetry add` updates *all* already installed dependencies. There should be way to disable this behavior.

Basically, dependencies upgrades should be controlled as much as possible. This behavior, in contrast, leads to absolutely uncontrolled upgrades. When someone *adds new* dependency, he/she definitely wants to only *add new* dependency, and not to *upgrade* existing. There is already designated way to upgrade "everything": just run `poetry update`.

The reason why there should be way to disable such behavior is as follows. Developing any real-world project ("more-than-hello-world") involves installing *a lot* of dependencies. For example, in the project I'm currently working on there is 70 direct dependencies (stated under `[tool.project.dependencies]`) and total 138 with deps-of-deps-of-deps (without development dependencies). When someone introduces new dependency to the project, he definitely doesn't want to stop his current work, examine all the upgrades coming in with this new package. Why to examine them, may you ask? Well, because upgrades should be *controlled*. And because *any* of these upgades could possible break the project. And this possibility increases with total number of dependencies. And by default poetry writes dependencies using "caret" notation, which in turn would be resulted in upgrades of minor (or even major, in case of deps-of-deps) versions of deps. And they could break the project...

So, options here may be:
- don't introduce new dependencies;
- manually check dependencies of new package, hack updated `poetry.lock` to include only those, then manually compute hash of `pyproject.toml` and update it accordingly in lock.

Both are no-option.

**Comment by abn** [comment]
@tribals this should not be the case at the moment, the installer is configured so that only specified dependencies and relevant parts of the dependency graph should be updated.

https://github.com/python-poetry/poetry/blob/b1422f1f5e7a40d737efafcb4f0fda47fd15b181/poetry/console/commands/add.py#L171

Can you provide an example `pyproject.toml` where this is happening? It should also be noted that there have been a few changes to how we are locking dependency graphs in `1.1` which will also make the situation in general better.

Additionally, while not related to the issue you have mentioned, the caret notation used by default only allows patch versions to be updated, which if the project is following semver should not introduce breaking changes.

**Comment by tribals** [comment]
Hmm. I tried to reproduce issue by myself in clean venv. And I failed. Seems like I did wrong flow:

- first, I added new dependency, it worked as expected - no extra-updates;
- then, I re-created venv for some reason (`poetry env remove & use`) and checked-out `poetry.lock` but kept `pyproject.toml` - with new dep included in it;
- and *then*, I tried to re-install everything with `poetry install`. It failed as expected - because of `pyproject.toml` & `poetry.lock` divergence;
- finally, I tried to fix that with `poetry lock` and then `poetry install` - and noticed extra-updates.

I do realise that doing `poetry add dep` and adding it into `pyproject.toml` and then doing `poetry lock` is not identical. But I misleadingly assumed that I came up with extra-updates after doing `poetry add dep`.

Sorry about opening issue.
